protocols/gossipsub/CHANGELOG.md

```diff
@@ -3,6 +3,12 @@
 - Deprecate `gossipsub::Config::idle_timeout` in favor of `SwarmBuilder::idle_connection_timeout`.
   See [PR 4648].
 
+<!-- Interal changes:
+
+- Allow new clippy lint.
+
+-->
```

Comment:
My new changelog lint forces us to always update the changelog. I'd suggest we just make an HTML comment in the changelogs for internal changes.

Thoughts?